pyOSPParser/system_configuration.py

```diff
@@ -0,0 +1,648 @@
+import json
+import os
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import List, Union, Dict
+
+import xmlschema
+
+PATH_TO_XML_SCHEMA = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    'xmlschema',
+    'OspSystemStructure.xsd'
+)
+
+
+class VariableType(Enum):
+    Real = 'Real'
+    Integer = 'Integer'
+    String = 'String'
+    Boolean = 'Boolean'
+
+
+class InterfaceType(Enum):
+    Variable = 'VariableConnection'
+    Signal = 'SignalConnection'
+    VariableGroup = 'VariableGroupConnection'
+    SignalGroup = "SignalGroupConnection"
+
+
+class InterfaceSubType(Enum):
+    Bond = 'BondConnection'
+    PlugSocket = 'PlugSocketConnection'
+    Scalar = 'ScalarConnection'
+    Sum = 'SumConnection'
+
+
+class OspSystemStructureAbstract(ABC):
+    @property
+    @abstractmethod
+    def _required_keys(self):
+        pass
+
+    def __init__(self, dict_xml: Union[Dict, None] = None, **kwargs):
+        if dict_xml:
+            self.from_dict_xml(dict_xml)
+        else:
+            for key in self._required_keys:
+                if key in kwargs:
+                    self.__setattr__(key, kwargs[key])
+                else:
+                    msg = "A required argument, '%s', is missing" % key
+                    raise TypeError(msg)
+            for key, value in kwargs.items():
+                if key not in self._required_keys:
+                    self.__setattr__(key, value)
+
+    @abstractmethod
+    def to_dict_xml(self):
+        """
+        This method should be defined for the inherited class
```

Comment:
@ abstractmethod already says that this should be defined in an implementation of the abstract base class.
Replace the comment with a comment on what the function should do.